packages/compartment-mapper/test/test-main.js

```diff
@@ -292,3 +294,106 @@ test('no transitive dev dependencies', async t => {
     },
   );
 });
+
+test('makeArchive / parseArchive, but with sha512 corruption', async t => {
+  t.plan(1);
+  await setup();
+
+  const { computeSha512 } = readPowers;
+
+  const corruptSha512 = bytes => {
+    const hash = computeSha512(bytes);
+    return `b4d${hash}`;
+  };
+
+  const archive = await makeArchive(
+    {
+      ...readPowers,
+      computeSha512: corruptSha512,
+    },
+    fixture,
+    {
+      modules,
+      dev: true,
+    },
+  );
+
+  const application = await parseArchive(archive, 'app.agar', {
+    computeSha512,
+  });
+
+  await t.throwsAsync(
+    () =>
+      application.import({
+        globals,
+        globalLexicals,
+        modules,
+        Compartment,
+      }),
+    {
+      message: /failed a SHA-512 integrity check/,
```

Comment:
Is it possible to test a hash mismatch in both the top-level compartment map, and in one of the modules being loaded, separately? Maybe by making the doctored hasher insert the corruption on the first call but not the later ones (and knowing exactly how many time the hasher is called during assembly). That's probably easier than creating a valid archive, taking it apart, flipping a bit in a hash, then reassembling it.